Gems/RemoteTools/Code/Source/RemoteToolsSystemComponent.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzFramework/Network/IRemoteTools.h>
+#include <AzNetworking/ConnectionLayer/IConnectionListener.h>
+#include <AzNetworking/DataStructures/ByteBuffer.h>
+#include <AzNetworking/Utilities/TimedThread.h>
+
+#include "Utilities/RemoteToolsJoinThread.h"
+#include "Utilities/RemoteToolsOutboxThread.h"
+
+namespace RemoteToolsPackets
+{
+    class RemoteToolsConnect;
+    class RemoteToolsMessage;
+} // namespace RemoteToolsPackets
+
+namespace RemoteTools
+{
+    // Slightly below AzNetworking's TCP max packet size for maximum message space with room for packet headers
+    static constexpr uint32_t RemoteToolsBufferSize = 16000;
+    using RemoteToolsMessageBuffer = AzNetworking::ByteBuffer<RemoteToolsBufferSize>;
+
+    struct RemoteToolsRegistryEntry
+    {
+        bool m_isHost;
+        AZ::Name m_name;
+        AzNetworking::IpAddress m_ip;
+
+        AzFramework::RemoteToolsEndpointContainer m_availableTargets;
+        AzFramework::RemoteToolsEndpointInfo m_lastTarget;
+        AZStd::vector<char, AZ::OSStdAllocator> m_tmpInboundBuffer;
```

Comment:
It would better clarify the intent that the package data is byte data by having the type be an `AZStd::vector<AZStd::byte>`.

Also why is the AZ::OSStdAllocator being used here.
There is now reason to not just use the normal `AZStd::allocator` adapter which uses the AZ::SystemAllocator by default.

```suggestion
        AZStd::vector<AZStd::byte> m_tmpInboundBuffer;
```